pkg/processor/trigger/v3iostream/event.go

```diff
@@ -41,3 +43,7 @@ func (e *Event) GetShardID() int {
 func (e *Event) GetOffset() int {
 	return int(e.record.SequenceNumber)
 }
+
+func (e *Event) GetTimestamp() time.Time {
+	return time.Unix(int64(e.record.ArrivalTimeSec), 0)
```

Comment:
As per [this](https://www.iguazio.com/docs/latest-release/services/data-layer/reference/web-apis/streaming-web-api/getrecords/#:~:text=contains%20these%20elements%3A-,ArrivalTimeSec,Type%3A%20Number,-SequenceNumber):
```suggestion
	return time.Unix(int64(e.record.ArrivalTimeSec), int64(e.record.ArrivalTimeNSec))
```